aten/src/ATen/SparseTensorImpl.h

```diff
@@ -80,6 +81,27 @@ struct SparseTensorImpl : public TensorImpl {
     }
     sparseDims_ = sparseDims;
     denseDims_ = denseDims;
+    #else
+    if (size.size() == 0) {
+      // An empty sparse tensor should be a 1-dimensional tensor of size [0],
+      // with sparseDims == 1 and denseDims == 0
+      size_ = {0};
+      sparseDims_ = 1;
+      denseDims_ = 0;
+    } else {
+      size_ = size;
+      sparseDims_ = sparseDims;
+      denseDims_ = denseDims;
+    }
+
+    // Check and preserve invariants
+    AT_ASSERT(sparseDims_ + denseDims_ == size_.size());
+    indices_.resize_({sparseDims_, values_.size(0)});
+    std::vector<int64_t> values_size = {values_.size(0)};
+    auto dense_size = sizes().slice(sparseDims_);
+    values_size.insert(values_size.end(), dense_size.begin(), dense_size.end());
+    values_.resize_(values_size);
```

Comment:
don't you need to set the other variables, like nnz, coalesced?  (I actually don't know why these weren't set previously).

Reply:
`raw_resize_` only changes the size of the sparse tensor and I think the number of non-zero elements and whether it's coalesced shouldn't be affected.

Reply:
`raw_resize` is a crap function and it requires too many caller invariants to be upheld to be used properly. Not necessarily for this PR but look for opportunities to use more safe versions of this function, like `resize_nnz`